### PR TITLE
Enhance documentation for cht-core automated tests

### PR DESCRIPTION
**Pull request by jkuester**
There was an existing page focused on "Fixing E2E Tests".  I have replaced this page with a more generic "Automated Tests" page that contains all the same information, but also has more broad sections about the tests in general.  I think we can continue to expand this page with more information in future PRs (e.g. I know we want to capture the general purpose for each type of tests and some decision support for how engineers can tell where they should add tests when making a code change...)!

One big question I did have is if we think that we should go ahead and pull all the contents from https://github.com/medic/cht-core/blob/master/TESTING.md into this page (and then just point to this page from that `TESTING.md` file)?

### CODE REVIEW

File: content/en/contribute/code/core/automated-tests.md [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+description: >
+  Automated testing for cht-core
+aliases: >
+  /core/guides/fixing-e2e-tests
+  /contribute/code/core/fixing-e2e-tests
+---
+
+# Automated Tests
+
+## Unit Tests
```
[the remark this answers is not in the document]
Comment (jkuester) [reply content/en/contribute/code/core/automated-tests.md:14]: Good idea!  I have added some basic working definitions for unit (and e2e) tests similar to what you suggested for the integration tests.  

Feel free to propose edits/additions!

**Review by lorerod — APPROVED** [review]

File: content/en/contribute/code/core/automated-tests.md [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+    e2e-tests o--Pouch/HTTPS--o cht-e2e
+```
+
+### Debugging E2E tests
+
+End to end (e2e) tests can be really difficult to debug - sometimes they fail seemingly at random, and sometimes they only fail on certain environments (eg: ci but not locally). This can make reproducing and reliably fixing the issue challenging, so here are some tips to help!
+
+#### Set the `DEBUG` flag
+
+Setting the `DEBUG` environment variable when running the tests locally will do the following:
```
Comment (tatilepizs) [inline content/en/contribute/code/core/automated-tests.md:93]:
```suggestion
Setting the `DEBUG` environment variable (e.g. `DEBUG=true npm run wdio-local`) when running the tests locally will do the following:
```

File: content/en/contribute/code/core/automated-tests.md [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+
+- Run the browser without the `headless` flag, so the browser will be displayed when running the tests
```
Comment (tatilepizs) [inline content/en/contribute/code/core/automated-tests.md:95]:
```suggestion
- Run the browser without the `headless` flag (details [here](https://github.com/medic/cht-core/blob/master/tests/e2e/wdio.conf.js#L86-L87)), so the browser will be displayed when running the tests
```

File: content/en/contribute/code/core/automated-tests.md [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+    integration-tests{Integration Tests}
+
+    integration-tests <--Pouch/HTTPS--> cht-e2e
+```
+
+## E2E Tests
+
+Our end-to-end tests are designed to test the entire system as a whole. They interact with the webapp as a user would, using [WebdriverIO](https://webdriver.io/) to control a headless browser session. They are not isolated from the rest of the system, and they do not use mocking.
+
+End-to-end tests are located in `tests/e2e`. Run them locally with the following:
```
Comment (tatilepizs) [inline content/en/contribute/code/core/automated-tests.md:53]:
```suggestion
End-to-end tests are located in [`tests/e2e`](https://github.com/medic/cht-core/tree/master/tests/e2e). Run them locally with the following:
```

File: content/en/contribute/code/core/automated-tests.md [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+
+End to end (e2e) tests can be really difficult to debug - sometimes they fail seemingly at random, and sometimes they only fail on certain environments (eg: ci but not locally). This can make reproducing and reliably fixing the issue challenging, so here are some tips to help!
+
+#### Set the `DEBUG` flag
+
+Setting the `DEBUG` environment variable when running the tests locally will do the following:
+
+- Run the browser without the `headless` flag, so the browser will be displayed when running the tests
+- Increase the test timeout from 2 minutes to 10 minutes
+- Prevent Mocha from automatically retrying tests that fail (by default a failing test is retried 5 times)
```
Comment (tatilepizs) [inline content/en/contribute/code/core/automated-tests.md:97]:
```suggestion
- Prevent Mocha from automatically retrying tests that fail (by default a failing test is retried 5 times, details [here](https://github.com/medic/cht-core/blob/master/tests/e2e/wdio.conf.js#L177))
```

**Review by tatilepizs — APPROVED** [review]
Thank you for this document @jkuester. I am sorry that took me so long to read it, but it is great and it contains really good information. I just added some links to the code that I think might be useful to understand better in case someone is a little confused 😊

**Comment by craig-landry** [comment]
@jkuester I appreciate the desire to have a page that handles testing more holistically. I currently am seeking to clarify unit, integration, and e2e testing expectations. I have markdown content prepared for guidance on this, but I also want to respect the effort to keep this PR moving and do more with more PRs:
> I think we can continue to expand this page with more information in future PRs

What is preferred:
1. Add content to this PR. I'll go make a commit on this branch.
2. Make a new branch from this branch and open a PR into this branch
3. Wait until this is merged and then open a new PR. This is basically the same as option 2 and a rebase depending on when this gets merged.
4. Something different?

**Comment by jkuester** [comment]
@tatilepizs, thanks for the thoughtful review! The links are super helpful!  

@craig-landry I am going to merge this, so you are free to just build a new PR on top of this one!  Sorry that it took so long to get this one landed!

**Pull request merged**